core/src/main/java/org/apache/accumulo/core/util/compaction/CompactionJobPrioritizer.java

```diff
@@ -19,42 +19,120 @@
 package org.apache.accumulo.core.util.compaction;
 
 import java.util.Comparator;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Objects;
 
+import org.apache.accumulo.core.clientImpl.Namespace;
+import org.apache.accumulo.core.data.NamespaceId;
+import org.apache.accumulo.core.data.TableId;
+import org.apache.accumulo.core.metadata.MetadataTable;
+import org.apache.accumulo.core.metadata.RootTable;
 import org.apache.accumulo.core.spi.compaction.CompactionJob;
 import org.apache.accumulo.core.spi.compaction.CompactionKind;
+import org.apache.accumulo.core.util.Pair;
+
+import com.google.common.base.Preconditions;
 
 public class CompactionJobPrioritizer {
 
   public static final Comparator<CompactionJob> JOB_COMPARATOR =
       Comparator.comparingInt(CompactionJob::getPriority)
           .thenComparingInt(job -> job.getFiles().size()).reversed();
 
-  public static short createPriority(CompactionKind kind, int totalFiles, int compactingFiles) {
+  private static final Map<Pair<TableId,CompactionKind>,Pair<Short,Short>> SYSTEM_TABLE_RANGES =
+      new HashMap<>();
+  private static final Map<Pair<NamespaceId,CompactionKind>,
+      Pair<Short,Short>> ACCUMULO_NAMESPACE_RANGES = new HashMap<>();
+
+  // Create ranges of possible priority values where each range has
+  // 2000 possible values. Give higher priority to tables where
+  // they have more files than allowed, user compactions over system
+  // compactions, root table over metadata table, metadata table over
+  // other system tables and user tables.
+  private static final Short TABLE_OVER_SIZE_MAX = Short.MAX_VALUE;
+  private static final Short ROOT_TABLE_USER_MAX = Short.MAX_VALUE - 2001;
+  private static final Short ROOT_TABLE_SYSTEM_MAX = (short) (ROOT_TABLE_USER_MAX - 2001);
+  private static final Short META_TABLE_USER_MAX = (short) (ROOT_TABLE_SYSTEM_MAX - 2001);
+  private static final Short META_TABLE_SYSTEM_MAX = (short) (META_TABLE_USER_MAX - 2001);
+  private static final Short SYSTEM_NS_USER_MAX = (short) (META_TABLE_SYSTEM_MAX - 2001);
+  private static final Short SYSTEM_NS_SYSTEM_MAX = (short) (SYSTEM_NS_USER_MAX - 2001);
+  private static final Short USER_TABLE_USER_MAX = (short) (SYSTEM_NS_SYSTEM_MAX - 2001);
+  private static final Short USER_TABLE_SYSTEM_MAX = 0;
+
+  static final Pair<Short,Short> TABLE_OVER_SIZE =
```

Comment:
Not sure if this would be an improvement.  Was wondering if creating a little inner class like the following and using that would clean up the code a bit over use Pair.

```java
static class PrioRange {
   final short min;
   final short max;
  PrioRange(short min, short max){this.min=min;this.max=max;}
}
```

Reply:
I guess it depends on what kind of improvement you are looking for. I did look at [Pair](https://commons.apache.org/proper/commons-math/javadocs/api-3.6.1/org/apache/commons/math3/util/Pair.html) in commons-math3. There is also [Range](https://commons.apache.org/proper/commons-lang/apidocs/org/apache/commons/lang3/Range.html) in commons-lang. I'm thinking the latter might be a good replacement here.

Reply:
The improvement I Was thinking of was just making the code easier to read.   When reviewing the code I was like ok so what does getFirst() mean, oh that means getMin().  So just thought that min and max terminology instead of getFirst() and getSecond() would make the code easier to read.

Reply:
The latest changes have getMinimum and getMaximum which is much easier to read.